valohai_cli/utils/ssh.py

```diff
@@ -0,0 +1,122 @@
+            return json.loads(event['message'][len(prefix):])
+    return None
+
+
+def select_key_from_folder(folder_path: str) -> Optional[str]:
+    if os.path.exists(folder_path):
+        ssh_pair_found, key_pairs = get_ssh_key_pairs_from_dir(folder_path)
+        if ssh_pair_found and len(key_pairs) == 1 and click.confirm(
+            f"Do you want to use {key_pairs[0][0]} as private key for SSH connection?"):
+            selected_private_key = key_pairs[0][0]
+        elif ssh_pair_found and len(key_pairs) > 1:
+            selected_private_key = prompt_for_key_selection(key_pairs)
+        else:
+            selected_private_key = None
+        return selected_private_key
+    else:
+        click.echo(f"No such directory found: {folder_path}")
+        return None
+
+
+def prompt_for_key_selection(key_pairs: list) -> Optional[str]:
+    # Display the list of key pairs with a numbered list
+    click.echo("Select the SSH key pair you want to use:")
+    for i, (private_key, public_key) in enumerate(key_pairs):
+        base_name = os.path.basename(private_key)
+        click.echo(f"{i + 1}: {base_name} (Private: {private_key}, Public: {public_key})")
+
+    choice = click.prompt("Enter the number of the key pair, enter -1 to select none of the above", type=int)
+    if choice == -1:
+        return None
+
+    if 1 <= choice <= len(key_pairs):
+        selected_private_key = key_pairs[choice - 1][0]
+        return selected_private_key
+    else:
+        click.echo("Invalid selection Please try again")
+        return None
+
+
+def get_default_ssh_folder() -> str:
+    home_dir = os.path.expanduser("~")
+    return os.path.join(home_dir, ".ssh")
+
+
+def get_ssh_key_pairs_from_dir(folder_path: str) -> tuple:
+    key_pairs = []
+    private_keys = {}
+    public_keys = {}
+
+    for filename in os.listdir(folder_path):
+        file_path = os.path.join(folder_path, filename)
+        if os.path.isdir(file_path):
+            continue
+        if filename.endswith(".pub"):
+            with open(file_path) as file:
+                first_line = file.readline().strip()
+                if first_line.startswith(("ssh-rsa", "ssh-dss", "ecdsa-sha2-", "ssh-ed25519")):
+                    base_name = filename[:-4]
+                    public_keys[base_name] = file_path
+        else:
+            with open(file_path) as file:
+                first_line = file.readline().strip()
+                if first_line.startswith("-----BEGIN ") and "PRIVATE KEY-----" in first_line:
+                    base_name = filename
+                    private_keys[base_name] = file_path
+
+    # Match private keys with public keys
+    for base_name, private_key in private_keys.items():
+        base_name = base_name.split(".")[0]
+        if base_name in public_keys:
+            key_pairs.append((private_key, public_keys[base_name]))
+
+    if key_pairs:
+        return True, key_pairs
+    else:
+        return False, []
```

Comment:
Since we kind of always need the public part to be around (to eventually send with the execution configuration), I think we could just try to find `.pub`s, then assume they'll be paired with a suffixless private key part. Also, let's not support `ssh-dss` keys, they're deprecated.

Something like:

```suggestion
    for pth in pathlib.Path(folder_path).glob("*.pub"):
        if not pth.is_file():
            continue
        try:
            with pth.open("rb") as f:
                if not first_line.startswith((b"ssh-rsa", b"ecdsa-sha2-", b"ssh-ed25519")):
                    continue
            priv_pth = pth.with_suffix("")
            if priv_pth.is_file():
                yield Keypair(public_path=pth, private_path=priv_pth)
        except OSError:  # e.g. unreadable key
            pass
```